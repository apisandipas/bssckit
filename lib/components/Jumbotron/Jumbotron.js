"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styledBreakpoints = require("styled-breakpoints");

var _theme = _interopRequireDefault(require("../../theme"));

var _themeFunctions = require("../../utils/themeFunctions");

var _base = require("../Utilities/base");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n  margin-bottom: 2rem;\n  background-color: ", ";\n\n  ", " {\n    padding: ", ";\n  }\n\n  ", " {\n    padding: ", ";\n  }\n\n  ", ";\n  ", ";\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n    border-radius: ", ";\n  "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n      border-radius: ", ";\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n      border-radius: ", ";\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n    padding: ", ";\n  "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n      padding: ", ";\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n      padding-right: ", ";\n      padding-left: ", ";\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var padding = function padding(props) {
  if (props.fluid) {
    return (0, _styledComponents.css)(_templateObject(), (0, _themeFunctions.getPadding)(props, 'jumbotron', 'fluidRight'), (0, _themeFunctions.getPadding)(props, 'jumbotron', 'fluidLeft'));
  } else if (props.pill) {
    return (0, _styledComponents.css)(_templateObject2(), (0, _themeFunctions.getPadding)(props, 'jumbotron', 'pill'));
  }

  return (0, _styledComponents.css)(_templateObject3(), (0, _themeFunctions.getPadding)(props, 'jumbotron', 'default'));
};

var borderRadius = function borderRadius(props) {
  if (props.fluid || props.noRadius) {
    return (0, _styledComponents.css)(_templateObject4(), (0, _themeFunctions.getBorderRadius)(props, 'jumbotron', 'noRadius'));
  } else if (props.pill) {
    return (0, _styledComponents.css)(_templateObject5(), (0, _themeFunctions.getBorderRadius)(props, 'jumbotron', 'pill'));
  }

  return (0, _styledComponents.css)(_templateObject6(), (0, _themeFunctions.getBorderRadius)(props, 'jumbotron', 'lg'));
};

var Jumbotron = (0, _styledComponents["default"])(_base.Div)(_templateObject7(), function (props) {
  return (0, _themeFunctions.getColor)(props, 'jumbotron', 'backgroundColor');
}, (0, _styledBreakpoints.up)('sm'), function (props) {
  if (props.pill) {
    return (0, _themeFunctions.getPadding)(props, 'jumbotron', 'lgPill');
  }

  return (0, _themeFunctions.getPadding)(props, 'jumbotron', 'sm');
}, (0, _styledBreakpoints.down)('md'), function (props) {
  if (props.pill) {
    return (0, _themeFunctions.getPadding)(props, 'jumbotron', 'smPill');
  }

  return (0, _themeFunctions.getPadding)(props, 'jumbotron', 'sm');
}, function (props) {
  return padding(props);
}, function (props) {
  return borderRadius(props);
});
Jumbotron.defaultProps = {
  theme: _theme["default"]
};
var _default = Jumbotron;
exports["default"] = _default;