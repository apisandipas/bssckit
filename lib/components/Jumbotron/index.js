"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Jumbotron", {
  enumerable: true,
  get: function get() {
    return _Jumbotron["default"];
  }
});

var _Jumbotron = _interopRequireDefault(require("./Jumbotron"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }