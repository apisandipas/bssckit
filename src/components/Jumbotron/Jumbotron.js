import styled, { css } from 'styled-components';
import { up, down } from 'styled-breakpoints';
import defaultTheme from '../../theme';
import { getColor, getPadding, getBorderRadius } from '../../utils/themeFunctions';
import { Div } from '../Utilities/base';

const padding = props => {
  if (props.fluid) {
    return css`
      padding-right: ${getPadding(props, 'jumbotron', 'fluidRight')};
      padding-left: ${getPadding(props, 'jumbotron', 'fluidLeft')};
    `;
  } else if (props.pill) {
    return css`
      padding: ${getPadding(props, 'jumbotron', 'pill')};
    `;
  }

  return css`
    padding: ${getPadding(props, 'jumbotron', 'default')};
  `;
};

const borderRadius = props => {
  if (props.fluid || props.noRadius) {
    return css`
      border-radius: ${getBorderRadius(props, 'jumbotron', 'noRadius')};
    `;
  } else if (props.pill) {
    return css`
      border-radius: ${getBorderRadius(props, 'jumbotron', 'pill')};
    `;
  }

  return css`
    border-radius: ${getBorderRadius(props, 'jumbotron', 'lg')};
  `;
};

const Jumbotron = styled(Div)`
  margin-bottom: 2rem;
  background-color: ${props => getColor(props, 'jumbotron', 'backgroundColor')};

  ${up('sm')} {
    padding: ${props => {
      if (props.pill) {
        return getPadding(props, 'jumbotron', 'lgPill');
      }
      return getPadding(props, 'jumbotron', 'sm');
    }};
  }

  ${down('md')} {
    padding: ${props => {
      if (props.pill) {
        return getPadding(props, 'jumbotron', 'smPill');
      }
      return getPadding(props, 'jumbotron', 'sm');
    }};
  }

  ${props => padding(props)};
  ${props => borderRadius(props)};
`;

Jumbotron.defaultProps = {
  theme: defaultTheme,
};

export default Jumbotron;
