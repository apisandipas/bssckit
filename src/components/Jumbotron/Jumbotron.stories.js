import React from 'react';
import { Container } from '../Grid';
import { Button } from '../Button';
import Jumbotron from './Jumbotron';

export default {
  title: 'Jumbotron',
};

export const simple = () => (
  <Container>
    <Jumbotron>
      <h1>Hello, world!</h1>
      <p>
        This is a simple hero unit, a simple jumbotron-style component for calling extra attention to featured
        content or information.
      </p>
      <hr />
      <p>
        It uses utility classes for typography and spacing to space content out within the larger container.
      </p>
      <Button primary lg>
        Learn more
      </Button>
    </Jumbotron>
  </Container>
);

export const fluid = () => (
  <Jumbotron fluid>
    <h1>Fluid jumbotron</h1>
    <p>This is a modified jumbotron that occupies the entire horizontal space of its parent.</p>
  </Jumbotron>
);
