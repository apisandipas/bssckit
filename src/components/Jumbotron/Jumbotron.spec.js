import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';

import { Jumbotron } from '../Jumbotron';

describe('Jumbotron Component', () => {
  it('matches Jumbotron', () => {
    const { container } = render(<Jumbotron />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches Jumbotron pill', () => {
    const { container } = render(<Jumbotron pill />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('border-radius', '10rem');
  });

  it('matches Jumbotron fluid', () => {
    const { container } = render(<Jumbotron fluid />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('padding-right', '0');
    expect(container.firstChild).toHaveStyleRule('padding-left', '0');
    expect(container.firstChild).toHaveStyleRule('border-radius', '0');
  });

  it('matches Jumbotron noRadius', () => {
    const { container } = render(<Jumbotron noRadius />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('border-radius', '0');
  });
});
