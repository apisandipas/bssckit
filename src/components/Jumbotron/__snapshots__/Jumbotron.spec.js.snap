// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jumbotron Component matches Jumbotron 1`] = `
.c0 {
  margin-bottom: 2rem;
  background-color: #f0f0f0;
  padding: 2rem 1rem;
  border-radius: 0.3rem;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (min-width:36em) {
  .c0 {
    padding: 4rem 2rem;
  }
}

@media (max-width:61.99875em) {
  .c0 {
    padding: 4rem 2rem;
  }
}

<div
  class="sc-fznyAO c0"
/>
`;

exports[`Jumbotron Component matches Jumbotron fluid 1`] = `
.c0 {
  margin-bottom: 2rem;
  background-color: #f0f0f0;
  padding-right: 0;
  padding-left: 0;
  border-radius: 0;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (min-width:36em) {
  .c0 {
    padding: 4rem 2rem;
  }
}

@media (max-width:61.99875em) {
  .c0 {
    padding: 4rem 2rem;
  }
}

<div
  class="sc-fznyAO c0"
/>
`;

exports[`Jumbotron Component matches Jumbotron noRadius 1`] = `
.c0 {
  margin-bottom: 2rem;
  background-color: #f0f0f0;
  padding: 2rem 1rem;
  border-radius: 0;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (min-width:36em) {
  .c0 {
    padding: 4rem 2rem;
  }
}

@media (max-width:61.99875em) {
  .c0 {
    padding: 4rem 2rem;
  }
}

<div
  class="sc-fznyAO c0"
/>
`;

exports[`Jumbotron Component matches Jumbotron pill 1`] = `
.c0 {
  margin-bottom: 2rem;
  background-color: #f0f0f0;
  padding: 2rem 8rem;
  border-radius: 10rem;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (min-width:36em) {
  .c0 {
    padding: 4rem 8rem;
  }
}

@media (max-width:61.99875em) {
  .c0 {
    padding: 4rem 4rem;
  }
}

<div
  class="sc-fznyAO c0"
/>
`;
